src/InstallationValidator.Core/Services/BatchComparisonReportingTask.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using InstallationValidator.Core.Assets;
+using InstallationValidator.Core.Domain;
+using OSPSuite.Core;
+using OSPSuite.Core.Reporting;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility;
+
+namespace InstallationValidator.Core.Services
+{
+   public interface IBatchComparisonReportingTask
+   {
+      Task CreateReport(BatchComparisonResult comparisonResult, string firstFolderPath, string secondFolderPath, bool openReport);
+   }
+
+   public class BatchComparisonReportingTask : IBatchComparisonReportingTask
+   {
+      private readonly IReportTemplateRepository _reportTemplateRepository;
+      private readonly IReportingTask _reportingTask;
+      private readonly IValidationLogger _validationLogger;
+      private readonly IApplicationConfiguration _applicationConfiguration;
+
+      public BatchComparisonReportingTask(IReportTemplateRepository reportTemplateRepository, IReportingTask reportingTask, IValidationLogger validationLogger, IApplicationConfiguration applicationConfiguration)
+      {
+         _reportTemplateRepository = reportTemplateRepository;
+         _reportingTask = reportingTask;
+         _validationLogger = validationLogger;
+         _applicationConfiguration = applicationConfiguration;
+      }
+
+      public async Task CreateReport(BatchComparisonResult comparisonResult, string firstFolderPath, string secondFolderPath, bool openReport = false)
+      {
+         var reportConfiguration = new ReportConfiguration
+         {
+            Template = _reportTemplateRepository.All().FirstOrDefault(),
+            Title = _applicationConfiguration.OSPSuiteNameWithVersion,
+            SubTitle = Assets.Reporting.FolderComparison,
+            ReportFile = reportOutputPath(firstFolderPath, DateTime.Now)
```

Comment:
I suggest to put the report in the NEW folder and not the OLD folder (typically first folder is existing result, second folders are new results)